### PR TITLE
LABS-1132 Move checkpoint metadata to shared storage & add an API to pick up a new checkpoint

### PR DESCRIPTION
**Pull request by pmacko86**
This PR makes the following changes:
* Stores the checkpoint metadata in shared storage using the functions introduced in LABS-1111.
* Adds an API for notifying the follower about a new checkpoint.
* Removes WiredTigerCheckpoints.txt and the oligarch metadata watcher thread, since they are no longer necessary.
* Updates the tests accordingly.

The API for picking up a checkpoint is through connection reconfiguration as explained in the ticket.

### CODE REVIEW

**Comment by quchenhao** [comment]
@pmacko86 Can we remove the dir store code then?

**Review by quchenhao — APPROVED** [review]
lgtm

**Comment by pmacko86** [comment]
> @pmacko86 Can we remove the dir store code then?

Unfortunately not yet, because PALM is still not working correctly (test_oligarch08 is still failing)...

File: src/conn/conn_oligarch.c [hunk 1]
```diff
@@ -8,192 +8,97 @@
...
-        if (F_ISSET(conn, WT_CONN_CLOSING))
+    /* Only followers are allowed to pick up new checkpoints. */
+    if (conn->oligarch_manager.leader)
+        WT_ERR(EINVAL);
+
+    /* Read the checkpoint metadata from the special metadata page. */
+    WT_ERR(__wt_disagg_get_meta(session, 0, checkpoint_id, item));
+
+    if (item->size >= sizeof(buf))
+        WT_ERR(EINVAL);
```
Comment (ddanderson) [inline src/conn/conn_oligarch.c:47]: Maybe ENOMEM would leave a better clue on failure.

File: src/conn/conn_oligarch.c [hunk 2]
```diff
@@ -8,192 +8,97 @@
...
+    memcpy(buf, item->data, item->size);
+    buf[item->size] = '\0';
+    if (item->size > 0 && buf[item->size - 1] == '\n')
+        buf[item->size - 1] = '\0';
+
+    /* Parse out the key and the new checkpoint config value. */
+    metadata_key = buf;
+    for (sep = (wt_off_t)item->size; sep >= 0; sep--)
```
Comment (ddanderson) [inline src/conn/conn_oligarch.c:55]: I'm wondering why this goes from the back to the front to search for '|'?  I would think that having a '|' would be much rarer than having a '|' in the value, so maybe front to back is better?

Also maybe '\n' is a better separator for this case? (could be done later)

File: src/conn/conn_oligarch.c [hunk 3]
```diff
@@ -8,192 +8,97 @@
...
 
-        /*
-         * WiredTiger will reload the dir store's checkpoint when opening a cursor: Opening a file
-         * cursor triggers __wt_btree_open (even if the file has been opened before).
-         */
-        WT_STAT_CONN_DSRC_INCR(session, oligarch_manager_checkpoints_refreshed);
-    }
+    /* Pull the value out. */
+    WT_ERR(md_cursor->get_value(md_cursor, &current_value));
+    len = strlen(metadata_value) + strlen("checkpoint=") + 1 /* for NUL */;
```
Comment (ddanderson) [inline src/conn/conn_oligarch.c:74]: Doesn't it need a comma, like: `strlen(",checkpoint=")` ?

File: src/conn/conn_oligarch.c [hunk 4]
```diff
@@ -946,9 +847,13 @@ __wti_disagg_conn_config(WT_SESSION_IMPL *session, const char **cfg, bool reconf
     bstorage = NULL;
     nstorage = NULL;
 
-    /* We don't currently allow reconfiguring disaggregated storage options. */
-    if (reconfig)
+    /* Reconfiguration can be used only to advance to the next checkpoint. */
+    if (reconfig) {
+        WT_ERR(__wt_config_gets(session, cfg, "disaggregated.checkpoint_id", &cval));
+        if (cval.len > 0 && cval.val >= 0)
+            WT_ERR(__disagg_pick_up_checkpoint(session, (uint64_t)cval.val));
```
Comment (ddanderson) [inline src/conn/conn_oligarch.c:854]: Should have an error if `(cval.len == 0 || cval.val < 0)`

**Review by ddanderson — APPROVED** [review]
LGTM - made some minor suggestions.

File: src/conn/conn_oligarch.c (hunk 3 above)
Comment (pmacko86) [reply src/conn/conn_oligarch.c:74]: The constructed string just starts with "checkpoint=", but I see it's confusing, so I will swap the order.

**Pull request merged**